packages/back-end/src/util/features.ts

```diff
@@ -346,6 +356,29 @@ export function getFeatureDefinition({
     ? revision.rules?.[environment] ?? settings.rules
     : settings.rules;
 
+  // If the feature has a holdout and it's enabled for the environment, add holdout as a
+  // pseudo force rule with a prerequisite condition. The environment being enabled is
+  // already checked in the getAllPayloadHoldouts function.
+  const holdoutRule: FeatureDefinitionRuleWithHoldout[] =
+    feature.holdout &&
+    holdoutsMap &&
+    holdoutsMap.get(feature.holdout.id)?.environmentSettings?.[environment]
+      ?.enabled
+      ? [
+          {
+            id: `fr_${uuidv4()}`,
```

Comment:
Use a deterministic hash of feature id and holdout id.  Also prefix with `holdout_` or something so we can identify it.